### PR TITLE
fix #7641: fix initial template load

### LINKED ISSUE #7641: [MU4 Issue] Template list starts with the wrong order
**Describe the bug**
New Score Wizard > Choose from template > Template list starts with the wrong order
In "Template" column random list of templatesIf you click on any item from Cathegory column and then return to "General" item from Cathegory list it will show the correct list

**To Reproduce**
1. Start New Score Wizard
2. Click on Choose from template
3. Look at the "Template" column

**Expected behavior**
It should display correct Template list from "General" Cathegory column from the startIt should be the sam

**Screenshots**
image
image (1)
2021-02-25 17_17_12-Window
e as in MU3

**Desktop (please complete the following information):**
MacOS
Windows

**Additional context**

### PR DESCRIPTION
**Pull request by qtweng**
Resolves: *https://github.com/musescore/MuseScore/issues/7641)*

*The load method loads all templates into visible templates, if we instead call updateTemplatesByCategory after generating the categories, it will load the relevant templates.*

- [x] I signed [CLA](https://musescore.org/en/cla)
- [x] I made sure the code in the PR follows [the coding rules](https://github.com/musescore/MuseScore/wiki/CodeGuidelines)
- [x] I made sure the code compiles on my machine
- [x] I made sure there are no unnecessary changes in the code
- [x] I made sure the title of the PR reflects the core meaning of the issue you are solving
- [x] I made sure the commit message(s) contain a description and answer the question "Why do those changes fix that particular issue?" or "Why are those changes really necessary as improvements?"
- [x] I made sure the commit message title starts with "fix #424242:" if there is a related issue
- [ ] I created the test (mtest, vtest, script test) to verify the changes I made

### CODE REVIEW

File: src/userscores/view/templatesmodel.cpp [hunk 1]
```diff
@@ -23,12 +23,11 @@ void TemplatesModel::load()
...
 
-    m_visibleTemplates = m_allTemplates;
-
     for (const Template& templ: m_allTemplates) {
         m_visibleCategoriesTitles << templ.categoryTitle;
     }
 
+    updateTemplatesByCategory();
     emit categoriesChanged();
     emit templatesChanged();
```
Comment (RomanPudashkin) [inline src/userscores/view/templatesmodel.cpp:32]: templatesChanged and currentTemplateChanged are already called from updateTemplatesByCategory, so you can remove these signals from TemplatesModel::load()

**Comment by RomanPudashkin** [comment]
@vshalkevich please, check this

**Comment by Jojo-Schmitz** [comment]
the 'merge' commit surely doesn't belong into this PR

**Comment by qtweng** [comment]
Whoops, let me fix that. Sorry, I am a beginner in open source contributions.

**Review by RomanPudashkin — APPROVED** [review]

**Comment by vshalkevich** [comment]
Can`t say that issue has been fixed. I still see the wrong Category order on create New Wizard Score page 
Here is how it should be 
изображение

**Comment by qtweng** [comment]
Ah, will further investigate.

**Comment by qtweng** [comment]
image
Looking at variable values, categories are inserted into HashSet and access seems to be random based on variations in the hash function.

**Comment by vshalkevich** [comment]
Yes it seems random even in the application

**Comment by qtweng** [comment]
@vshalkevich I think this should fix it.

**Comment by vshalkevich** [comment]
now it seems fixed

**Review by RomanPudashkin — APPROVED** [review]

**Pull request merged**